harness/determined/common/experimental/trial.py

```diff
@@ -157,6 +157,7 @@ def __init__(self, trial_id: int, session: api.Session):
 
         self.hparams: Optional[Dict[str, Any]] = None
         self.summary_metrics: Optional[Dict[str, Any]] = None
+        self.status: Optional[bindings.experimentv1State] = None
```

Comment:
might be worth making a separate public-facing status object. would basically be duplicated code, but i don't like that we're surfacing bindings to users.

Reply:
I'm interested here in what @rb-determined-ai thinks about a

```python
from determined.common.api.bindings import experimentv1State
```

and then using that.

Reply:
I am of the opinions that swagger-generated bindings generally suck and are unergonomic, making them undesirable to ever give to end users.  They're basically only for type safety, and I think we should be careful not to leak them to end users.

Specifically in the enum case, proto always generates an UNSPECIFIED enum value which makes sense in the protobuf serialization layer but semantically does not make sense to expose in python.  If UNSPECIFIED were necessary, we should be accepting `Optional[thing]` where `None` serves the purpose of UNSPECIFIED.